Gemfile

```diff
@@ -25,6 +25,9 @@ gem 'jbuilder', '~> 2.7'
 # Use Active Storage variant
 # gem 'image_processing', '~> 1.2'
 
+# Controller Testing
+gem 'rails-controller-testing'
```

Comment:
What are your thought on just testing using Requests and Model testing? And not testing the controller directly.

Reply:
I'm aware that Aaron has asked us to do controller testing, but I'm asking anyway. :)

Reply:
For some reason I'm not too comfortable with using just request  + model tests without involving the controller as it seems like the test isn't enough. How do I know the right template is being called? When a redirect occurs, how do I know it's showing the right view? I guess I can address template rendering with End-to-End testing - ex: Capybara, so this is really a non-issue.
But then again, I think if it was an API only project, the request + model tests would do just fine as I'll be returning json anyways?

Reply:
Yes and yes... Yes I agree you need Capybara and yes once you use it for your E2E testing then it resolves the controller testing issue. Personally I like that, because it reduces my test load, but not (I believe) it's effectiveness. Interesting insight about the API, because I hadn't thought of that and it's worth considering before these kinds of decisions are made for sure. Thanks for sharing!!! I just wanted to have a discussion with you.